utilities.tex

```diff
@@ -787,6 +677,111 @@
 two \cv-unqualified non-reference types if at least one of them depends on a
 user-defined type. Such a specialization need not have a member named
 \tcode{type}.
+
+\rSec1[meta.type.synop.rng]{Header \tcode{<experimental/ranges/type_traits>} synopsis}
+\begin{codeblock}
+namespace std { namespace experimental { namespace ranges { inline namespace v1 {
+  // \ref{meta.unary.prop.rng}, type properties:
+  template <class T, class U> struct is_swappable_with;
+  template <class T> struct is_swappable;
+
+  template <class T, class U> struct is_nothrow_swappable_with;
+  template <class T> struct is_nothrow_swappable;
+
+  template <class T, class U> constexpr bool is_swappable_with_v
+    = is_swappable_with<T, U>::value;
+  template <class T> constexpr bool is_swappable_v
+    = is_swappable<T>::value;
+
+  template <class T, class U> constexpr bool is_nothrow_swappable_with_v
+    = is_nothrow_swappable_with<T, U>::value;
+  template <class T> constexpr bool is_nothrow_swappable_v
+    = is_nothrow_swappable<T>::value;
+}}}}
+\end{codeblock}
+
+\rSec2[meta.unary.prop.rng]{Additional type properties}
+
+\pnum
+These templates provide access to some of the more important properties of types.
+
+\pnum
+It is unspecified whether the library defines any full or partial specializations
+of any of these templates.
+
+\pnum
+For all of the class templates \tcode{X} declared in this subclause, instantiating
+that template with a template argument that is a class template specialization may
+result in the implicit instantiation of the template argument if and only if the
+semantics of \tcode{X} require that the argument must be a complete type.
+
+\pnum
+For the purpose of defining the templates in this subclause, a function call
+expression \tcode{declval<T>()} for any type \tcode{T} is considered to be a
+trivial~(\cxxref{basic.types}, \cxxref{special}) function call that is not an
+odr-use~(\cxxref{basic.def.odr}) of \tcode{declval} in the context of the
+corresponding definition notwithstanding the restrictions of~(\cxxref{declval}).
+
+\begin{libreqtab3b}{Additional type property predicates}{tab:type-traits.properties}
+\\ \topline
+\lhdr{Template} &   \chdr{Condition}    &   \rhdr{Precondition}    \\ \capsep
+\endfirsthead
+\continuedcaption\\
+\topline
+\lhdr{Template} &   \chdr{Condition}    &   \rhdr{Precondition}    \\ \capsep
+\endhead
+
+\indexlibrary{\idxcode{is_swappable_with}}%
+\tcode{template <class T, class U>}\br
+  \tcode{struct is_swappable_with;} &
+  The expressions \tcode{ranges::swap(declval<T>(),}\brk\tcode{ declval<U>())}
```

Comment:
When I built this locally, the `ranges::swap(declval<T>(),` overran the middle column and partially obscured the text in the right column.